api/client/container/update.go

```diff
@@ -135,14 +135,21 @@ func runUpdate(dockerCli *client.DockerCli, opts *updateOptions) error {
 
 	ctx := context.Background()
 
-	var errs []string
+	var (
+		warns []string
+		errs  []string
+	)
 	for _, container := range opts.containers {
-		if err := dockerCli.Client().ContainerUpdate(ctx, container, updateConfig); err != nil {
+		if r, err := dockerCli.Client().ContainerUpdate(ctx, container, updateConfig); err != nil {
+			warns = append(warns, r.Warnings...)
 			errs = append(errs, err.Error())
 		} else {
```

Comment:
I think in this place, we still need `warns = append(warns, r.Warnings...)`, since when err is nil, r.Warnings still can be non-nil. Correct me if I am wrong. @runcom